File: VB/CustomDrawButtonEvent/DrawButtonEventArgs.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.Utils.Drawing
Imports System.Drawing
Imports DevExpress.Utils

Namespace CustomDrawButtonEvent

	Public Class DrawButtonEventArgs
		Public Sub New(ByVal info As EditorButtonObjectInfoArgs)
			Me.info = info
			handled_Renamed = False
		End Sub

		Private info As EditorButtonObjectInfoArgs
		Private handled_Renamed As Boolean


		Public Property Handled() As Boolean
			Get
				Return handled_Renamed
			End Get
			Set(ByVal value As Boolean)
				If handled_Renamed <> value Then
					handled_Renamed = value
				End If
			End Set
		End Property
		Public Property State() As ObjectState
			Get
				Return info.State
			End Get
			Set(ByVal value As ObjectState)
				If info.State <> value Then
					info.State = value
				End If
			End Set
		End Property

		Public ReadOnly Property Appearance() As AppearanceObject
			Get
				Return info.Appearance
			End Get
		End Property
		Public ReadOnly Property Index() As Integer
			Get
				Return info.Button.Index
			End Get
		End Property
		Public ReadOnly Property Caption() As String
			Get
				Return info.Button.Caption
			End Get
		End Property
		Public ReadOnly Property Graphics() As Graphics
			Get
				Return info.Graphics
			End Get
		End Property
		Public ReadOnly Property Cache() As GraphicsCache
			Get
				Return info.Cache
			End Get
		End Property
		Public ReadOnly Property Bounds() As Rectangle
			Get
				Return info.Bounds
			End Get
		End Property
		Public ReadOnly Property Image() As Image
			Get
				Return info.Button.Image
			End Get
		End Property
		Public ReadOnly Property Kind() As ButtonPredefines
			Get
				Return info.Button.Kind
			End Get
		End Property
		Public ReadOnly Property ImageLocation() As ImageLocation
			Get
				Return info.Button.ImageLocation
			End Get
		End Property
	End Class

End Namespace

File: CS/CustomDrawButtonEvent/DrawButtonEventArgs.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Drawing;
using DevExpress.Utils.Drawing;
using System.Drawing;
using DevExpress.Utils;

namespace CustomDrawButtonEvent {

    public class DrawButtonEventArgs {
        public DrawButtonEventArgs(EditorButtonObjectInfoArgs info) {
            this.info = info;
            handled = false;
        }

        private EditorButtonObjectInfoArgs info;
        private bool handled;


        public bool Handled { 
            get {
                return handled;
            }
            set {
                if (handled != value)
                    handled = value;
            }
        }
        public ObjectState State { 
            get {
                return info.State;
            }
            set {
                if (info.State != value) {
                    info.State = value;
                }
            }
        }

        public AppearanceObject Appearance { get { return info.Appearance; } }
        public int Index { get { return info.Button.Index; } }
        public string Caption { get { return info.Button.Caption; } }
        public Graphics Graphics { get { return info.Graphics; } }
        public GraphicsCache Cache { get { return info.Cache; } }
        public Rectangle Bounds { get { return info.Bounds; } }
        public Image Image { get { return info.Button.Image; } }
        public ButtonPredefines Kind { get { return info.Button.Kind; } }
        public ImageLocation ImageLocation { get { return info.Button.ImageLocation; } }
    }

}
